### PR TITLE
Add notes describing why groff 1.22.3 must be used

### PR DESCRIPTION
**Pull request by kevinburkemeter**
Updates #3387.
Updates #3391.

I will abide by the [code of conduct](https://github.com/rubygems/rubygems/blob/master/CODE_OF_CONDUCT.md).

### CODE REVIEW

File: bundler/doc/documentation/WRITING.md [hunk 1]
```diff
@@ -44,6 +44,32 @@ $ man man/bundle-cookies.1
...
+   installed version is 1.22.3) and `bsdmainutils` (for the `col` program).
+
+2. Download `groff` and compile version 1.22.3 from source. You can download
+   groff from here: https://savannah.gnu.org/git/?group=groff. Check out git tag
+   `1.22.3`, run `./configure && make && make install` and then rerun groff.
+
+    Note, on a Mac, the default installed `groff` is version 1.19, and the
+    Homebrew version of `groff` is 1.22.4, both of which will generate incorrect
+    output. You must also set `GROFF_NO_SGR=true` in the environment in order to
+    avoid escape sequences being printed into the text file.
```
Comment (deivid-rodriguez) [inline bundler/doc/documentation/WRITING.md]: I think this is what `col -b` fixes so maybe it's unnecesary?

Comment (kevinburkemeter) [reply bundler/doc/documentation/WRITING.md]: Nope still needed it with `-b` enabled. Without GROFF_NO_SGR, on Macs a bunch of escape sequences like `022m` get printed.

Comment (deivid-rodriguez) [reply bundler/doc/documentation/WRITING.md]: Ah ok, thanks! Does the `-c` flag also work for that, or there's no such flag on the MacOS version?

Comment (kevinburkemeter) [reply bundler/doc/documentation/WRITING.md]: I tried the "-c" flag but still getting escape sequences in the output without GROFF_NO_SGR set.

Comment (deivid-rodriguez) [reply bundler/doc/documentation/WRITING.md]: Interesting. Then let's document the thing that actually works :+1:.

**Comment by deivid-rodriguez** [comment]
This PR _might_  no longer be needed as per https://github.com/rubygems/rubygems/pull/3391#issuecomment-600262796.

**Comment by deivid-rodriguez** [comment]
Actually, we should still document the discrepancy about control codes in MacOS, but hopefully no need to mention 1.22 vs 1.23 discrepancies.

**Comment by kevinburkemeter** [comment]
Oh! Fantastic, then. I'll update the PR later today.

**Comment by deivid-rodriguez** [comment]
> Oh! Fantastic, then. I'll update the PR later today.

Great, thanks!

**Comment by deivid-rodriguez** [comment]
Could you update this one, @kevinburkemeter?

**Comment by deivid-rodriguez** [comment]
Closing due to lack of feedback. Feel free to reopen and update the PR if you still want to do this :+1:.

**Pull request closed without merging**